In [ ]:
pip install kaggle


   ---------------------------------------- 0/3 [kagglesdk]
   ---------------------------------------- 0/3 [kagglesdk]
   ---------------------------------------- 0/3 [kagglesdk]
   ---------------------------------------- 0/3 [kagglesdk]
   ------------- -------------------------- 1/3 [jupytext]
   ------------- -------------------------- 1/3 [jupytext]
   -------------------------- ------------- 2/3 [kaggle]
   ---------------------------------------- 3/3 [kaggle]

Note: you may need to restart the kernel to use updated packages.


Superstore Dataset Analysis

CRISP-DM Framework - Prisca AI Intersnship

Business Understanding: Identify which segments, categories, and regions drive profit vs loss in the Superstore data, and spot patterns in discounting that hurt margins.

In [2]:
import pandas as pd

df = pd.read_csv("./data/Sample - Superstore.csv", encoding ="latin")
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


Data Understanding (Phase 2)

Checking the size, structure, and quality of the dataset before cleaning.

In [5]:
print(df.shape)
print(df.dtypes)

(9994, 21)
Row ID             int64
Order ID             str
Order Date           str
Ship Date            str
Ship Mode            str
Customer ID          str
Customer Name        str
Segment              str
Country              str
City                 str
State                str
Postal Code        int64
Region               str
Product ID           str
Category             str
Sub-Category         str
Product Name         str
Sales            float64
Quantity           int64
Discount         float64
Profit           float64
dtype: object


In [5]:
df.tail()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
9989,9990,CA-2014-110422,2014-01-21,2014-01-23,Second Class,TB-21400,Tom Boeckenhauer,Consumer,United States,Miami,...,33180,South,FUR-FU-10001889,Furniture,Furnishings,Ultra Door Pull Handle,25.248,3,0.2,4.1028
9990,9991,CA-2017-121258,2017-02-26,2017-03-03,Standard Class,DB-13060,Dave Brooks,Consumer,United States,Costa Mesa,...,92627,West,FUR-FU-10000747,Furniture,Furnishings,Tenex B1-RE Series Chair Mats for Low Pile Car...,91.960,2,0.0,15.6332
9991,9992,CA-2017-121258,2017-02-26,2017-03-03,Standard Class,DB-13060,Dave Brooks,Consumer,United States,Costa Mesa,...,92627,West,TEC-PH-10003645,Technology,Phones,Aastra 57i VoIP phone,258.576,2,0.2,19.3932
9992,9993,CA-2017-121258,2017-02-26,2017-03-03,Standard Class,DB-13060,Dave Brooks,Consumer,United States,Costa Mesa,...,92627,West,OFF-PA-10004041,Office Supplies,Paper,"It's Hot Message Books with Stickers, 2 3/4"" x 5""",29.600,4,0.0,13.3200
9993,9994,CA-2017-119914,2017-05-04,2017-05-09,Second Class,CC-12220,Chris Cortes,Consumer,United States,Westminster,...,92683,West,OFF-AP-10002684,Office Supplies,Appliances,"Acco 7-Outlet Masterpiece Power Center, Wihtou...",243.160,2,0.0,72.9480


In [7]:
print(df.isnull().sum())


Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64


In [8]:
print(df.duplicated().sum())

0


Data Preparation (Phase 3)

Dataset had no missing values or duplicates. Key prep step:
- Converted Order Date and Ship Date from text to datetime, so date-based calculations would be possible if needed later.

(Note: I considered cretaing extra features like Profit Margin and Shipping Duration, but decided the original columns were enough to answer my business question, so I focused analysis on Sales, Profit, Discount, Category, Region, and Segment instead.)

In [17]:
df["Order Date"] = pd.to_datetime(df["Order Date"])
df["Ship Date"] = pd.to_datetime(df["Ship Date"])

print(df.dtypes)

Row ID                    int64
Order ID                    str
Order Date       datetime64[us]
Ship Date        datetime64[us]
Ship Mode                   str
Customer ID                 str
Customer Name               str
Segment                     str
Country                     str
City                        str
State                       str
Postal Code               int64
Region                      str
Product ID                  str
Category                    str
Sub-Category                str
Product Name                str
Sales                   float64
Quantity                  int64
Discount                float64
Profit                  float64
dtype: object


In [6]:
df.describe()

,Row ID,Order Date,Ship Date,Postal Code,Sales,Quantity,Discount,Profit
count,9994.000000,9994,9994,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000
mean,4997.500000,2016-04-30 00:07:12.259355,2016-05-03 23:06:58.571142,55190.379428,229.858001,3.789574,0.156203,28.656896
min,1.000000,2014-01-03 00:00:00,2014-01-07 00:00:00,1040.000000,0.444000,1.000000,0.000000,-6599.978000
25%,2499.250000,2015-05-23 00:00:00,2015-05-27 00:00:00,23223.000000,17.280000,2.000000,0.000000,1.728750
50%,4997.500000,2016-06-26 00:00:00,2016-06-29 00:00:00,56430.500000,54.490000,3.000000,0.200000,8.666500
75%,7495.750000,2017-05-14 00:00:00,2017-05-18 00:00:00,90008.000000,209.940000,5.000000,0.200000,29.364000
max,9994.000000,2017-12-30 00:00:00,2018-01-05 00:00:00,99301.000000,22638.480000,14.000000,0.800000,8399.976000
std,2885.163629,NaN,NaN,32063.693350,623.245101,2.225110,0.206452,234.260108


In [18]:
print(df["Category"].unique())
print(df["Sub-Category"].unique())
print(df["Region"].unique())
print(df["Segment"].unique())

<StringArray>
['Furniture', 'Office Supplies', 'Technology']
Length: 3, dtype: str
<StringArray>
[  'Bookcases',      'Chairs',      'Labels',      'Tables',     'Storage',
 'Furnishings',         'Art',      'Phones',     'Binders',  'Appliances',
       'Paper', 'Accessories',   'Envelopes',   'Fasteners',    'Supplies',
    'Machines',     'Copiers']
Length: 17, dtype: str
<StringArray>
['South', 'West', 'Central', 'East']
Length: 4, dtype: str
<StringArray>
['Consumer', 'Corporate', 'Home Office']
Length: 3, dtype: str


Cleaning Summary

- No missing values or duplicate rows found
- Order Date and Ship Date converted from text to datetime
- Numeric columns (Sales, Quantity, Discount, Profit) checked 
- No invalid values found
- Categorical columns checked for spelling / consistency - all clean
- Dataset required minimal cleaning; ready for analysis

In [ ]:
total_sales = df["Sales"].sum()
print("Total Sales:", total_sales)

Total Sales: 2297200.8603


In [9]:
total_profit = df["Profit"].sum()
print("Total Profit:", total_profit)

Total Profit: 286397.0217


In [10]:
overall_margin = total_profit / total_sales
print("Overall Profit Margin:", overall_margin)

Overall Profit Margin: 0.12467217240315605


Finding: Overall profit margin is approximately 12.5% - across all sales combined, about 12.5 cents of every $1 in revenue became profit. This is an average acress the whole dataset, not a fixed rate for every product - some categories (like Technology) perform above this average, while others (like Furniture) fall well below it, as explored later in this analysis.

In [12]:
total_orders = df["Order ID"].nunique()
print("Total orders:", total_orders)

Total orders: 5009


In [13]:
sales_by_region = df.groupby("Region")["Sales"].sum()
print(sales_by_region)

Region
Central    501239.8908
East       678781.2400
South      391721.9050
West       725457.8245
Name: Sales, dtype: float64


In [14]:
profit_by_region = df.groupby("Region")["Profit"].sum()
print(profit_by_region)

Region
Central     39706.3625
East        91522.7800
South       46749.4303
West       108418.4489
Name: Profit, dtype: float64


Finding: Sales vs Profit by Region

- West: $725,458 sales, $108,418 profit
- East: $678,781 sales, $91,523 profit
- Central: $501,240 sales, $39,706 profit
- South: $391,722 sales, $46,749 profit

WEST is the strongest performer overall - highest sales AND highest profit. 

Interesting anomaly: Central has higher sales than South ($501k vs $392k), but lower profit ($39.7k vs $46.7k). This suggests Central may be over - discounting or selling a less profitable product mix - worth investigating further (checking average discount by region).


In [16]:
discount_by_region = df.groupby("Region")["Discount"].mean()
print(discount_by_region)

Region
Central    0.240353
East       0.145365
South      0.147253
West       0.109335
Name: Discount, dtype: float64


Confirmed: Central has the highest average discount (24.0%) of all regions, 
nearly double West's (10.9%). This strongly suggests aggressive discounting 
is the reason Central sells more than South but keeps far less profit.

In [18]:
sales_by_category = df.groupby("Category")["Sales"].sum()
print(sales_by_category)

Category
Furniture          741999.7953
Office Supplies    719047.0320
Technology         836154.0330
Name: Sales, dtype: float64


In [19]:
profit_by_category = df.groupby("Category")["Profit"].sum()
print(profit_by_category)

Category
Furniture           18451.2728
Office Supplies    122490.8008
Technology         145454.9481
Name: Profit, dtype: float64


Finding: Category Performance

- Furniture: $741,999 sales, $18,451 profit (appproximately 2.5% margin)
- Office Supplies: $719,047 sales, $122,491 profit (approximately 17.0% margin)
- Technology: $836,154 sales, $145,455 profit (approximately 17.4% margin)

Furniture has sales comparable to Office Supplies, but profit 6-7x lower. 
Its margin (~2.5%) is far below Office Supplies and Technology (~17% each), 
suggesting heavy discounting or high costs specific to furniture items. 
Worth drilling into Sub-Category next to find exactly which products 
are the problem.

In [20]:
subcategory_profit = df.groupby("Sub-Category")[["Sales", "Profit"]].sum()
print(subcategory_profit)

                    Sales      Profit
Sub-Category                         
Accessories   167380.3180  41936.6357
Appliances    107532.1610  18138.0054
Art            27118.7920   6527.7870
Binders       203412.7330  30221.7633
Bookcases     114879.9963  -3472.5560
Chairs        328449.1030  26590.1663
Copiers       149528.0300  55617.8249
Envelopes      16476.4020   6964.1767
Fasteners       3024.2800    949.5182
Furnishings    91705.1640  13059.1436
Labels         12486.3120   5546.2540
Machines      189238.6310   3384.7569
Paper          78479.2060  34053.5693
Phones        330007.0540  44515.7306
Storage       223843.6080  21278.8264
Supplies       46673.5380  -1189.0995
Tables        206965.5320 -17725.4811


In [21]:
df[df["Sub-Category"] == "Bookcases"]["Category"].unique()

<StringArray>
['Furniture']
Length: 1, dtype: str

In [ ]:
subcategory_profit = df.groupby("Sub-Category")[["Sales", "Profit"]].sum()
subcategory_profit_sorted = subcategory_profit.sort_values("Profit")
print(subcategory_profit_sorted)

                    Sales      Profit
Sub-Category                         
Tables        206965.5320 -17725.4811
Bookcases     114879.9963  -3472.5560
Supplies       46673.5380  -1189.0995
Fasteners       3024.2800    949.5182
Machines      189238.6310   3384.7569
Labels         12486.3120   5546.2540
Art            27118.7920   6527.7870
Envelopes      16476.4020   6964.1767
Furnishings    91705.1640  13059.1436
Appliances    107532.1610  18138.0054
Storage       223843.6080  21278.8264
Chairs        328449.1030  26590.1663
Binders       203412.7330  30221.7633
Paper          78479.2060  34053.5693
Accessories   167380.3180  41936.6357
Phones        330007.0540  44515.7306
Copiers       149528.0300  55617.8249


Finding: Loss-making Sub-categories

When i sorted sub-categories by profit, three came out negative:
- Tables: -$17,725 (the biggest loss, even though it had $206,965 in sales)
- Bookcases: -$3,472
- Supplies: -$1,189

Tables and Bookcases are both part of the Furniture category, which explains why Furniture's overall margin (~2.5%) was so much lower than Office Supplies and Technology earlier. Tables especially stands out - it sells a lot, but still loses money on most sales, which suggests it might be discounted too heavily compared to what it actually costs.

In [24]:
segment_summary = df.groupby("Segment")[["Sales", "Profit"]].sum()
print(segment_summary)

                    Sales       Profit
Segment                               
Consumer     1.161401e+06  134119.2092
Corporate    7.061464e+05   91979.1340
Home Office  4.296531e+05   60298.6785


In [25]:
top_products = df.groupby("Product Name")["Profit"].sum().sort_values(ascending=False).head(10)
print(top_products)

Product Name
Canon imageCLASS 2200 Advanced Copier                                          25199.9280
Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind     7753.0390
Hewlett Packard LaserJet 3310 Copier                                            6983.8836
Canon PC1060 Personal Laser Copier                                              4570.9347
HP Designjet T520 Inkjet Large Format Printer - 24" Color                       4094.9766
Ativa V4110MDD Micro-Cut Shredder                                               3772.9461
3D Systems Cube Printer, 2nd Generation, Magenta                                3717.9714
Plantronics Savi W720 Multi-Device Wireless Headset System                      3696.2820
Ibico EPK-21 Electric Binding System                                            3345.2823
Zebra ZM400 Thermal Label Printer                                               3343.5360
Name: Profit, dtype: float64


In [27]:
bottom_products = df.groupby("Product Name")["Profit"].sum().sort_values().head(10)
print(bottom_products)

Product Name
Cubify CubeX 3D Printer Double Head Print                           -8879.9704
Lexmark MX611dhe Monochrome Laser Printer                           -4589.9730
Cubify CubeX 3D Printer Triple Head Print                           -3839.9904
Chromcraft Bull-Nose Wood Oval Conference Tables & Bases            -2876.1156
Bush Advantage Collection Racetrack Conference Table                -1934.3976
GBC DocuBind P400 Electric Binding System                           -1878.1662
Cisco TelePresence System EX90 Videoconferencing Unit               -1811.0784
Martin Yale Chadless Opener Electric Letter Opener                  -1299.1836
Balt Solid Wood Round Tables                                        -1201.0581
BoxOffice By Design Rectangular and Half-Moon Meeting Room Tables   -1148.4375
Name: Profit, dtype: float64


Finding: Top and Bottom Products

Looking at individual produts, the  Canon imageCLASS 2200 Advanced Copier was the clear top performer btinging in $25,199 profit — well ahead of every other single product.

On the losing end, the  Cubify CubeX 3D Printer (Double Head) lost the most money at $-8,879, and its Triple Head version  also lost money (-$3,839), which makes me think this entire 3D printer product line is unprofitable.

I also noticed several conference/meeting tables (Chromcraft, Bush Advantage, Balt, BoxOffice) showing up among the worst performers  - which lines up with Tables being the most loss-making sub-category overall.

Summary of Findings

Overall, the Superstore made about $2.3 million in sales, but only $286,397 of that turned into actual profit - so roughly 12.5% of every dollar sold was kept as profit, and therest went to costs.

Breaking it down further, a few things stood out to me:

1. Central region discounts too much. Central gives an average discount of 24%, way higher than every other region (West is only 11%). This is probably why Central has higher sales than South but ends up with less profit - they're giving away too much on each sale.

2. Furniture does'nt make sense financially. Furniture sells almost as much as Office Supplies, but makes 6-7x less profit. When I looked closer, it's mainly two sub-categories - Tables and Bookcases - that are actually losing money, not just making low profit.

3. A couple of specific products are the real problem. The Cubify CubeX 3D Printer (both th double-head and triple-head versions) lost the most money out of every single product. I also noticed several conference tables showing up as losses, which matches the Tables issue above.

4. Everything else (segments, other categories) looked normal - sales and profit moved together the way you'd expect, nothing alarming.

Comparing this to what Prisca AI does

I did this analysis by hand - pandas, one cell at a time, following CRISP-DSM. Based on what I've seen on Prisca AI's own site, the platform is built to automate a lot of these same steps: it reviews your data structure, flags inconsistencies, suggests cleaning strategies, and can even engineer new features on its own (like pulling out a month/year column or deriving a metric I'd otherwise write pandas code for). It also auto-builds dashboards and has a forecasting tool built in.

Doing this manually meant I made every cleaning decision myself and could trace a sinding all the way down - e.g going from "Central's profit looks low" to "it's because of a 24% average discount rate" to "specifically Tables and the Cubify 3D printers are the biggest losses." That kind of step-by-step drill-down is exactly the "human control" piece Prisca AI says it's designed to keep, even while automating the busywork.

Where i think the platform would genuinely save time: the cleaning checks I did (missing values, duplicates, date formatting, feature creation) sound like exactly what it automates in seconds. Where manual work stilll mattered: deciding which business questions to ask and why a pattern was happening - that reasoning is the part I'd want to double check even if a tool automated the machine.